content/_code-samples/build-a-wallet/js/1_hello.js

```diff
@@ -0,0 +1,71 @@
+const { app, BrowserWindow } = require('electron')
+
+const path = require('path')
+const xrpl = require("xrpl")
+
+const testnetUrl = "wss://s.altnet.rippletest.net:51233"
+
+/**
+ * This function creates a WebService client, which connects to the XRPL and fetches the latest ledger index.
+ *
+ * @returns {Promise<number>}
+ */
+const getValidatedLedgerIndex = async () => {
+    const client = new xrpl.Client(testnetUrl)
+
+    await client.connect()
+
+    const ledgerRequest = {
+        "command": "ledger",
+        "ledger_index": "validated"
+    }
+
+    const ledgerResponse = client.request(ledgerRequest)
+
+    await client.disconnect()
+
+    return (await ledgerResponse).result.ledger_index
```

Comment:
Why do we disconnect before awaiting the ledger response? I feel like that could create a race condition where the connection closes and so you don't hear a response.